packages/client/hmi-client/src/components/widgets/tera-chart-settings-panel.vue

```diff
@@ -45,7 +45,7 @@
 				</div>
 				<section v-if="isColorPickerEnabled">
 					<h6>Color Picker</h6>
-					<input type="color" :value="primaryColor" @change="onColorChange($event)" />
+					<input type="color" :value="activeSettings?.primaryColor ?? ''" @change="onColorChange($event)" />
```

Comment:
I think this component is simple enough so that it doesn't need to rely on `useChartSettings` composable. I feel it's just fine to use props and event emitter and handle updates in parent component with the composable.  I don't see much benefits to bring the whole composable which includes other unnecessary stuff here.

Reply:
Then that `primaryColor` should never be null or undefined. `activeSetting` as a prop should be as consumable as possible. for this simple component

Reply:
`primaryColor` is an optional property and it can be undefined by design.